app/Http/Controllers/Staff/DonationSubscriptionController.php

```diff
@@ -0,0 +1,160 @@
+     */
+    public function edit(Request $request, int $id): \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+    {
+        $user = $request->user();
+        abort_unless($user->group->is_admin, 403);
+
+        $curDate = Carbon::now();
+        $vip_sub = DonationSubscription::findOrFail($id);
+
+        return view('Staff.donations.subscriptions.edit', [
+            'vip_sub' => $vip_sub,
+            'curdate' => $curDate,
+            'date'    => $curDate
+        ]);
+    }
+
+    /**
+     * Save a VIP Subscription change.
+     */
+    public function update(Request $request, int $id): \Illuminate\Http\RedirectResponse
+    {
+        $user = $request->user();
+        abort_unless($user->group->is_admin, 403);
+
+        $curDate = Carbon::now();
+        $vip_sub_end_at_old = "";
+        $vip_sub = DonationSubscription::findOrFail($id);
+
+        // Set Dates from Input form
+        $vip_sub->start_at = $request->input('start_at');
+        $vip_sub->end_at = $request->input('end_at');
+
+        $v = validator($vip_sub->toArray(), [
+            'start_at' => 'required|date',
+            'end_at'   => 'required|date|after:start_at',
+        ]);
+
+        if ($v->fails()) {
+            return redirect()->route('staff.donations.subscriptions.index')
+                ->withErrors($v->errors());
+        }
+        $vip_sub->save();
+
+        return redirect()->route('staff.donations.subscriptions.index')
+            ->withSuccess('VIP Subscription Was Updated Successfully!');
+    }
+
+    /**
+     * VIP Add Form.
+     */
+    public function create(): \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+    {
+        return view('Staff.donations.subscriptions.create');
+    }
+
+    /**
+     * Store A New VIP.
+     */
+    public function store(Request $request): \Illuminate\Http\RedirectResponse
+    {
+        $user = $request->user();
+        abort_unless($user->group->is_admin, 403);
+
+        $vip = new DonationSubscription();
+        $vipUsername = $request->input('username');
+        // Get Users ID to store FK in DB
+        $vip->user_id = User::query()
+            ->select(['id'])
+            ->where('username', '=', $vipUsername)
+            ->value('id');
+        $vip->start_at = $request->input('start_at');
+        $vip->end_at = $request->input('end_at');
+        $vip->donation_item_id = $request->input('donation_item_id');
+        $vip->is_gifted = 0;
+        $vip->is_active = 0;
+
+        $v = validator($vip->toArray(), [
+            'user_id'  => 'required',
+            'start_at' => 'required|date',
+            'end_at'   => 'required|date',
+        ]);
+
+        if ($v->fails()) {
+            return to_route('staff.donations.subscriptions.index')
+                ->withErrors($v->errors());
+        }
+        $vip->save();
+
+        return to_route('staff.donations.subscriptions.index')
+            ->withSuccess('New VIP Subscription added!');
+    }
```

Comment:
```suggestion
    public function store(Request $request): \Illuminate\Http\RedirectResponse
    {
        $userId = User::query()
            ->select(['id'])
            ->where('username', '=', $request->username)
            ->value('id');

        $validated = $request->validate([
            'donation_item_id' => [
                'exists:donation_items,id',
            ],
            'username'  => [
                'exists:users,username',
                'exclude',
            ],
            'start_at' => [
                'required',
                'date',
            ],
            'end_at'   => [
                'required',
                'date'
            ],
        ]);
        
        DonationSubscription::create([
            'user_id'   => $userId,
            'is_gifted' => false,
            'is_active' => false,
        ] + $validated));

        return to_route('staff.donations.subscriptions.index')
            ->withSuccess('New VIP Subscription added!');
    }
```

Move the validation array to a form request.